src/Common/src/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -228,10 +229,15 @@ public override ComputedStaticFieldLayout ComputeStaticFieldLayout(DefType defTy
             return result;
         }
 
-        private ref StaticsBlock GetStaticsBlockForField(ref ComputedStaticFieldLayout layout, FieldDesc field)
+        protected virtual ref StaticsBlock GetStaticsBlockForField(ref ComputedStaticFieldLayout layout, FieldDesc field)
         {
             if (field.IsThreadStatic)
-                return ref layout.ThreadStatics;
+            {
```

Comment:
"Not my business" but shouldn't we be rather conservative in keeping the CoreRT behavior unchanged and just modifying the R2R algorithm to cater for the thread static base distinction? [I haven't still finished the complete review so I apologize in advance if I'm just misunderstanding something.]

Reply:
OK, so this is the "default below CoreRT". If that's fine to be based on CoreCLR statics placement algorithm, I don't have any objections.

Reply:
My reasoning was, if we're going to put 4 tables in the type system the fundamental default better be to use them all, and have CoreRT be the weird case that collapses two tables into one. If Michal feels strongly against this way I can flip the default. In particular, we'll need to go adjust the `NetNativeFieldLayoutAlgorithm` for UTC-based compilation.